### ISSUE TITLE
spanner: failed to NewClient WithCredentialsJSON, might be caused by newBuiltinMetricsTracerFactory

### ISSUE THREAD

**Issue by RainbowDashy** [googleapis/google-cloud-go#11194]
## Client

spanner
## Environment

go 1.23.2
## Code and Dependencies

```go
package spanner_test

import (
	"context"
	"log"
	"testing"

	"cloud.google.com/go/spanner"
	"google.golang.org/api/option"
)

func TestConnect(t *testing.T) {
	ctx := context.Background()
	j := "<service account key in json format>"
	_, err := spanner.NewClient(ctx, "projects/foo/instances/bar/databases/d", option.WithCredentialsJSON([]byte(j)))
	if err != nil {
		log.Fatal(err)
	}
}

```

<details>
  <summary>go.mod</summary>

```text
module modname

go 1.23.2

require (
   	cloud.google.com/go/spanner v1.71.0
)
```

</details>

## Expected behavior

I expect no error. But I get the error 
`credentials: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information`.

With some debugging, I found out that the error is returned by `newBuiltinMetricsTracerFactory` at L510.
https://github.com/googleapis/google-cloud-go/blob/f06a55a5680e588241315d760b95d9402452a288/spanner/client.go#L510

It seems to me that the problem is `newBuiltinMetricsTracerFactory` doesn't respect the `WithCredentialsJSON` option.

**Comment by GavinFrazar** [comment]
I ran into this in test code as well and worked around it by just disabling metrics:

```go
	clientCfg := spanner.ClientConfig{
		DisableNativeMetrics: true,
		SessionPoolConfig:    spanner.DefaultSessionPoolConfig,
	}
	clt, err := spanner.NewClientWithConfig(ctx, databaseID, clientCfg, opts...)
```

**Comment by rahul2393** [comment]
@RainbowDashy @GavinFrazar Can you please check if v1.73.0 already fixes this?
PR: https://github.com/googleapis/google-cloud-go/pull/11109

**Comment by GavinFrazar** [comment]
> @RainbowDashy @GavinFrazar Can you please check if v1.73.0 already fixes this? PR: #11109

I can confirm that v1.73.0 fixed my tests without disabling the metrics
